.github/workflows/lint.yml

```
@@ -0,0 +1,19 @@
+on: [pull_request]
+name: changed files lint
+jobs:
+  scan:
+    name: npm lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+          fetch-tags: true
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22.11.0'
+      - name: install dependencies
+        run: npm ci
+      - name: npm lint
+        run: npx eslint $(git diff origin/${{ github.event.pull_request.base.ref }} --name-only | xargs)
```

Requires origin prefix because the base ref does not include the fully qualified name of the ref as required by git diff.

---

```suggestion
        run: npx eslint $(git diff "origin/$GITHUB_REF" --name-only | xargs)
        env:
                GITHUB_REF: ${{ github.event.pull_request.base.ref }}
```

Consider pulling out `${{ github.event.pull_request.base.ref }}` into an environment variable to avoid a template injection.

---

I see why the failure is happening, `GITHUB_REF` is already an environment variable: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables

I would switch it to `GITHUB_BASE_REF` or something similar.